### PR TITLE
Restaurant

### PR DESCRIPTION
**Pull request by mkhajure7000**

### CODE REVIEW

File: app/controllers/restaurants_controller.rb [hunk 1]
```diff
@@ -1,5 +1,46 @@
 class RestaurantsController < ApplicationController
+  before_action :require_restaurant, only: [:update, :edit,:destroy]
```
Comment (Shubham-Porwall) [inline app/controllers/restaurants_controller.rb]: use %i

File: app/controllers/restaurants_controller.rb [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+    redirect_to restaurants_path, status: :see_other
+  end
+
+  private
+  def restaurant_params
+    params.require(:restaurant).permit(:name, :address)
+  end
+
+  def require_restaurant
+    @restaurant = current_user.restaurants.find(params[:id])
```
Comment (Shubham-Porwall) [inline app/controllers/restaurants_controller.rb]: handle negative scenarios

File: test/fixtures/restaurants.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (Shubham-Porwall) [inline test/fixtures/restaurants.yml]: remove unnecessary files

File: app/controllers/restaurants_controller.rb (hunk 2 above)
Comment (Shubham-Porwall) [reply app/controllers/restaurants_controller.rb]: ^^

File: app/controllers/restaurants_controller.rb [hunk 4]
```diff
@@ -1,5 +1,50 @@
...
+    if @restaurant.save
+      redirect_to restaurants_path
+    else
+      flash[:register_errors] = @restaurant.errors.full_messages
+      redirect_to new_restaurant_path
+    end
+  end
+    
+  def edit
+    if @restaurant.nil?
```
Comment (Shubham-Porwall) [inline app/controllers/restaurants_controller.rb]: You can handle this condition in filter so it will help's for all three actions !

**Review by Shubham-Porwall — APPROVED** [review]
Looks Good to me !

File: app/controllers/restaurants_controller.rb [hunk 5]
```diff
@@ -1,5 +1,49 @@
...
+    @restaurant.destroy
+    redirect_to restaurants_path, status: :see_other
+  end
+
+  private
+  def restaurant_params
+    params.require(:restaurant).permit(:name, :address)
+  end
+
+  def require_restaurant
```
Comment (kalpana19sharma) [inline app/controllers/restaurants_controller.rb]: Rename this method

File: app/controllers/restaurants_controller.rb [hunk 6]
```diff
@@ -1,5 +1,49 @@
...
+    @restaurant = current_user.restaurants.find_by(id: params[:id])
+    if @restaurant.nil?
```
Comment (kalpana19sharma) [inline app/controllers/restaurants_controller.rb]: prefer using `.present?` or `.blank?`

File: app/views/restaurants/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<div class="w-25 container">
+  <%= form_with model: @restaurant do |form| %>
+    <%if @restaurant.errors.any? %>
+      <div class="alert alert-danger">
+        <% @restaurant.errors.full_messages.each do |message| %>
+          <div><%= message %></div>
+        <% end %>
+      </div>
```
Comment (kalpana19sharma) [inline app/views/restaurants/_form.html.erb]: Extract the error rendering logic in a shared partial. I think we have same logic for users views as well.

File: app/views/restaurants/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    <% end %>
+    <%= form.label :restaurant_name %><br>
+    <div class="mb-3">
+      <%= form.text_field :name, class: "form-control"%>
+    </div>
+    <%= form.label :Address %><br>
+    <div class="mb-3">
+      <%= form.text_field :address, class: "form-control"%>
```
Comment (kalpana19sharma) [inline app/views/restaurants/_form.html.erb]: Fix spaces

File: app/views/restaurants/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+<h1>Update Restaurant</h1>
+<%=render "form", restaurant: @restaurant %>
```
Comment (kalpana19sharma) [inline app/views/restaurants/edit.html.erb]: Fix spaces

File: app/views/restaurants/index.html.erb [hunk 10]
```diff
@@ -1 +1,17 @@
 <h1>Welcome Admin</h1>
+<h1><%= link_to "Add Restaurant", new_restaurant_path, class: "btn btn-success" %></h1>
+<h1><%= button_to "logout", {:controller => :sessions, :action => 'destroy', :id => current_user.id }, :method => :delete, class: "btn btn-success"%></h1>
```
Comment (kalpana19sharma) [inline app/views/restaurants/index.html.erb]: Don't use arrow syntax

File: app/views/restaurants/index.html.erb [hunk 11]
```diff
@@ -1 +1,17 @@
...
+<div style="text-align:center;"> 
+  <% @restaurants.each do |restaurant| %>
+    <div class="card" style="width: 20rem; text-align:center;display:inline-block;">
+      <img class="card-img-top" src="#" alt="Card image cap" height="200" width="200" style="border-radius:.60rem;">
```
Comment (kalpana19sharma) [inline app/views/restaurants/index.html.erb:7]: don't use inline css

File: app/views/users/index.html.erb [hunk 12]
```diff
@@ -1,8 +1,6 @@
 <% if current_user.present? %>
   <h1>Welcome user</h1>
-  <%= link_to "log out", session_path(current_user.id), data: {
-                    turbo_method: :delete
-                  } %>
+  <h1><%= button_to "logout", {:controller => :sessions, :action => 'destroy', :id => current_user.id }, :method => :delete, class: "btn btn-success"%></h1>
```
Comment (kalpana19sharma) [inline app/views/users/index.html.erb]: Don't use arrow syntax

Comment (kalpana19sharma) [reply app/views/users/index.html.erb]: And fix spaces

File: config/routes.rb [hunk 13]
```diff
@@ -3,7 +3,7 @@
   root to: "users#index"
   resources :users, only: %i[index new create]
   resources :sessions, only: %i[new create destroy]
-  resources :restaurants, only: [:index]
+  resources :restaurants
```
Comment (kalpana19sharma) [inline config/routes.rb]: are you using show action?

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Pull request merged**